crates/svm-runtime/src/runtime/ptr.rs

```diff
@@ -0,0 +1,56 @@
+use std::ffi::c_void;
+use std::ops::{Deref, DerefMut};
+
+use svm_types::Type;
+
+use crate::Runtime;
+
+/// A Smart-pointer for a `Runtime`
+///
+/// Its main usage will be FFI related code.
+#[repr(C)]
+pub struct RuntimePtr {
+    inner: Box<dyn Runtime>,
+}
+
+static RUNTIME_PTR_TY: Type = Type::of::<RuntimePtr>();
+
+impl RuntimePtr {
```

Comment:
`svm_byte_array` could have implemented `std::ops::Deref / std::ops::DerefMut` but I don't think it would be a good usage since it's pretty generic.

`svm_byte_array` saves a contiguous memory region of bytes.
Since these bytes can be virtually anything, I've added the `type_id` so that we can attach metadata about the original Rust type as part of the resources tracking.